pkg/controller/controller_utils_test.go

```diff
@@ -406,24 +414,105 @@ func TestSortingActivePods(t *testing.T) {
 	pods[8].Status.ContainerStatuses = []v1.ContainerStatus{{RestartCount: 2}, {RestartCount: 1}}
 	pods[8].CreationTimestamp = then
 
-	getOrder := func(pods []*v1.Pod) []string {
-		names := make([]string, len(pods))
-		for i := range pods {
-			names[i] = pods[i].Name
+	expected := getPodNamesInOrder(pods)
+
+	for i := 0; i < 20; i++ {
+		idx := rand.Perm(numPods)
+		randomizedPods := make([]*v1.Pod, numPods)
+		for j := 0; j < numPods; j++ {
+			randomizedPods[j] = pods[idx[j]]
 		}
-		return names
+		sort.Sort(ActivePods(randomizedPods))
+		actual := getPodNamesInOrder(randomizedPods)
+
+		assert.EqualValues(t, expected, actual, "expected %v, got %v", expected, actual)
 	}
+}
 
-	expected := getOrder(pods)
+func TestSortingActivePodsWithRanks(t *testing.T) {
```

Comment:
This isn't really what I want to see for a sort test.  Create a table driven test with input scenarios in a concise form (helpers to generate each entry) and have each test check various edge cases.

Sort is a very easy thing to get wrong and random permutations don't build confidence, reviewable use cases do.

Reply:
Actually it's even better if you just test Less and have your table test do an initial setup, then test both forwards and backwards calls with Less (and they should either be equal or reversed).  I.e.:

```
test {
  a: something,
  b: something_else,
  val: true,
}

for ... {
   set := ActivePodsWithRanks{items: []...{test.a, test.b}}
   if set.Less(0, 1) != test.value {
      // error
   }
   if set.Less(1, 0) == test.value {
     // error
   }
}
```

that confines the test even more and demonstrates Less satisfies the sort interface (it's really easy in these complicated Less() functions to accidentally miss a reversible case which makes your sort unstable)

Reply:
I've now changed `TestSortingActivePodsWithRanks` to a table-driven test similar to what you describe.  I omitted `val` and instead defined each test case in terms of a lesser value and a greater value—I hope I didn't miss some nuance there, and I hope I covered enough edge cases.

Reply:
Oh, I misunderstood what you meant about "reversible cases", but I think I understand now—I have added some test cases to make sure that `Less` returns false given two equal pods.